qiskit/pulse/schedule/pulse_schedule.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+Schedule.
+"""
+import logging
+from abc import ABCMeta, abstractmethod
+from typing import List, Union
+
+from qiskit.pulse.channels import PulseChannel, ChannelBank
+from qiskit.pulse.commands import PulseCommand, FunctionalPulse, SamplePulse
+from qiskit.pulse.exceptions import ScheduleError
+
+logger = logging.getLogger(__name__)
+
+
+class TimedPulseBlock(metaclass=ABCMeta):
+    """
+    Common interface of TimedPulse and PulseSchedule (Component in the Composite Pattern)."""
+
+    @abstractmethod
+    def start_time(self) -> int:
+        pass
+
+    @abstractmethod
+    def end_time(self) -> int:
+        pass
+
+    @abstractmethod
+    def duration(self) -> int:
+        pass
+
+    @abstractmethod
+    def children(self) -> List['TimedPulseBlock']:
+        pass
+
+
+class TimedPulse(TimedPulseBlock):
+    """TimedPulse = Pulse with start time context."""
+
+    def __init__(self, pulse_command: PulseCommand, to_channel: PulseChannel, start_time: int):
+        if isinstance(pulse_command, to_channel.__class__.supported):
+            self.command = pulse_command
+            self.channel = to_channel
+            self.t0 = start_time
+        else:
+            raise ScheduleError("%s (%s) is not supported on %s (%s)" % (
+                                pulse_command.__class__.__name__, pulse_command.name,
+                                to_channel.__class__.__name__, to_channel.name))
+
+    def start_time(self) -> int:
+        return self.t0
+
+    def end_time(self) -> int:
+        return self.t0 + self.command.duration
+
+    def duration(self) -> int:
+        return self.command.duration
+
+    def children(self) -> List[TimedPulseBlock]:
+        return None
+
+
+class PulseSchedule(TimedPulseBlock):
```

Comment:
I  think this should just be `Schedule`, and in general, we should remove the name `Pulse` from API classes. The classes are already part of the `pulse` module namespace so adding it to the class is redundant.